### PR TITLE
Fix issue with collapse_metrics #45

### PR DESCRIPTION
**Pull request by gemygk**
Added ncRNA models to the CDS discard list (cds.fasta.lengths)

### CODE REVIEW

**Comment by gemygk** [comment]
@cschu , can you please check. The test works fine now.

File: minos/scripts/calculate_cdslen.py [hunk 1]
```diff
@@ -14,8 +14,20 @@ def read_fasta(f):
 			yield sid, "".join(seq)
 
 	with open(_out, "w") as fout:
-		for sid, seq in read_fasta(_in):
+		cdna_set = set()
+		for sid, seq in read_fasta(_in_cdna):
+			if sid not in cdna_set:
+				cdna_set.add(sid)
```
Comment (cschu) [inline minos/scripts/calculate_cdslen.py:20]: the `if` is not necessary

can just do : `cdna_set = set(sid for sid, seq in read_fasta(_in_cdna))`

File: minos/scripts/calculate_cdslen.py [hunk 2]
```diff
@@ -14,8 +14,20 @@ def read_fasta(f):
...
+		cds_set = set()
+		for sid, seq in read_fasta(_in_cds):
+			if sid not in cds_set:
```
Comment (cschu) [inline minos/scripts/calculate_cdslen.py:23]: s. above

File: minos/scripts/calculate_cdslen.py [hunk 3]
```diff
@@ -14,8 +14,20 @@ def read_fasta(f):
...
+				cds_set.add(sid)
 			seq = seq.upper().replace("U", "T")
 			has_stop = any(seq.endswith(stop_codon) for stop_codon in ("TAA", "TGA", "TAG"))
 			discard = (has_stop and len(seq) < min_cds_length) or (not has_stop and len(seq) < min_cds_length - 3)
 			print(sid, int(discard), sep="\t", flush=True, file=fout)
+
+		# get the ncRNA models and add them to CDS discard list
+		for sid in cdna_set - cds_set:
+			discard = True
+			print(sid, int(discard), sep="\t", flush=True, file=fout)
```
Comment (cschu) [inline minos/scripts/calculate_cdslen.py:33]: here you can also just do `print(sid, 1, ...)` as the `discard` variable is always `True`

**Review by cschu — COMMENTED** [review]
minor suggestions :)

File: minos/scripts/calculate_cdslen.py (hunk 1 above)
Comment (gemygk) [reply minos/scripts/calculate_cdslen.py:20]: I agree

File: minos/scripts/calculate_cdslen.py (hunk 3 above)
Comment (gemygk) [reply minos/scripts/calculate_cdslen.py:33]: I did the way suggested first, but to keep it consistent with the code above I made it explicit.

Comment (cschu) [reply minos/scripts/calculate_cdslen.py]: fair enough. it is not like this would cause a performance issue in this case :D

**Comment by gemygk** [comment]
Thanks @cschu I have done the changes.

**Pull request merged**